cypress-io/esbuild
# fix: various issues with sourcemaps

Helps with https://github.com/cypress-io/cypress/issues/22983

### User facing changelog


Sourcemaps now point to the right line when generated via `snapshot`

### Additional details


There were several issues that needed fixing:

* We need to ignore any of the generated code in terms of source maps. This is most easily determined based on the indent level. All generated code is included inside of functions so they should have an indent level of at least 1.
* We need to ensure that lines are updated even we are doing a "norewrite" snapshot.

Note: This repo needs some work in terms of making it more official (PR templates, automated builds/releases), but that is going to be addressed with: https://github.com/cypress-io/cypress/issues/22981

### Steps to test


I added tests for this to ensure that generated sourcemaps can be followed back to the original files: `verify-snapshot-source-maps`

### How has the user experience changed?


Stack traces will point to the right place now

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
